packages/smooth_app/lib/pages/product/edit_product_page.dart

```diff
@@ -57,201 +62,222 @@ class _EditProductPageState extends State<EditProductPage> {
         if (refreshedProduct != null) {
           _product = refreshedProduct;
         }
-        final Brightness brightness = Theme.of(context).brightness;
+        final ThemeData theme = Theme.of(context);
+        final Brightness brightness = theme.brightness;
+        final Size screenSize = MediaQuery.of(context).size;
 
         return SmoothScaffold(
           appBar: AppBar(
-            title: AutoSizeText(
-              getProductName(_product, appLocalizations),
-              maxLines: 2,
+            title: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                AutoSizeText(
+                  getProductName(_product, appLocalizations),
+                  maxLines: _barcodeVisibleInAppbar ? 1 : 2,
+                ),
+                if (_product.barcode?.isNotEmpty == true)
+                  Visibility(
```

Comment:
What about using `AnimatedSwitcher` instead of the `Visibility` widget? It should make the transition animated.

Reply:
Since the title can initially be on two lines, the rendering is not well appealed.
A custom one is required here. We should maybe implement it later?